library/common/http/dispatcher.cc

```diff
@@ -477,19 +477,38 @@ const LowerCaseString ClusterHeader{"x-envoy-mobile-cluster"};
 const std::string BaseCluster = "base";
 const std::string BaseWlanCluster = "base_wlan";
 const std::string BaseWwanCluster = "base_wwan";
+const LowerCaseString H2UpstreamHeader{"x-envoy-mobile-upstream-protocol"};
```

Comment:
I propose simplifying this with a `x-envoy-mobile-force-h2` header. If a header with this name is present assume inferred h2 support, if not, default to h1.

Reply:
I had originally written this PR with that approach, and actually with something similar to what you suggest below re: central configuration. 

However, I moved to this approach for a few reasons:
1.  The enum allows us to express potential expansion of future options: h3, auto (when alpn support lands on upstream connections).
2. The enum allows us to more easily migrate existing code to those future options without breaking the API if we find ourselves in a future where we care more strictly about API breakage as, hopefully, we see more adopters.
3. IMO the protocol support of your destination is an application level concern (if the library can't do protocol negotiation for you), and the application developer should tell us that prior knowledge. I am also not a big fan of centralized configuration given at boot time, as you know :)

Right now absence of the header defaults us to http1 per the implementation in the Http::Dispatcher::setDestinationCluster. I can do:

> I'd suggest moving this out of the constructor, and adding it via a builder method.

per your suggestion make this setting a less strong requirement. I think this brings us closer to the sentiment of making this engine-level configuration without having to wire more configuration through.

lmk what you think.